pkg/controller/s3/bucket/notificationConfig.go

```diff
@@ -331,3 +292,54 @@ func (in *NotificationConfigurationClient) CreateOrUpdate(ctx context.Context, b
 func (*NotificationConfigurationClient) Delete(_ context.Context, _ *v1beta1.Bucket) error {
 	return nil
 }
+
+// LateInitialize is responsible for initializing the resource based on the external value
+func (in *NotificationConfigurationClient) LateInitialize(ctx context.Context, bucket *v1beta1.Bucket) error {
+	external, err := in.client.GetBucketNotificationConfigurationRequest(&awss3.GetBucketNotificationConfigurationInput{Bucket: awsclient.String(meta.GetExternalName(bucket))}).Send(ctx)
+	if err != nil {
+		return awsclient.Wrap(err, notificationGetFailed)
+	}
+	if emptyConfiguration(external) {
+		// There is nothing to initialize from AWS
+		return nil
+	}
+
+	in.logger.Debug(fmt.Sprintf("called LateInitialize for %s", reflect.TypeOf(in).Elem().Name()))
+
+	config := bucket.Spec.ForProvider.NotificationConfiguration
+	if config == nil {
+		// We need the configuration to exist so we can initialize
+		bucket.Spec.ForProvider.NotificationConfiguration = &v1beta1.NotificationConfiguration{}
+		config = bucket.Spec.ForProvider.NotificationConfiguration
+	}
+
+	// A list is provided by AWS
+	if external.LambdaFunctionConfigurations != nil {
+		if config.LambdaFunctionConfigurations == nil {
```

Comment:
```suggestion
	if len(external.LambdaFunctionConfigurations) != 0 && len(config.LambdaFunctionConfigurations) == 0 {
```

We can late-init only in this case.